core/player/src/binding/__tests__/index.test.ts

```diff
@@ -162,6 +162,16 @@ describe("errors", () => {
     );
   });
 
+  test("works for bindings with escaped numeric bindings", () => {
+    const parser = new BindingParser({
+      get: () => {
+        return true;
+      },
+    });
+
+    expect(parser.parse("foo['01'].baz").asString()).toBe("foo.01.baz");
```

Comment:
Good question. Current behavior is that would parsed as an integer and we should probably keep that behavior. that does mean the parsing of that binding needs to probably happen a bit differently than it is now because the context of if a segment was wrapped in the escape syntax doesn't make it down to the binding constructor.

Reply:
👍 let's make sure we have tests for `foo.01.baz` and `foo[01].baz` to make sure we maintain existing behavior to be safe